arraycontext/impl/pytato/compile.py

```diff
@@ -399,7 +422,12 @@ def __call__(self, arg_id_to_arg) -> ArrayContainer:
         evt.wait()
 
         def to_output_template(keys, _):
-            return self.actx.thaw(out_dict[self.output_id_to_name_in_program[keys]])
+            name_in_program = self.output_id_to_name_in_program[keys]
+            return self.actx.thaw(to_tagged_cl_array(
+                out_dict[name_in_program],
+                axes=get_cl_axes_from_pt_axes(
+                    self.name_in_program_to_axes[name_in_program]),
```

Comment:
Could store the CL axes instead, to avoid this conversion cost.

Reply:
Not without some type-incoherencies, recorded it over here: https://github.com/inducer/pytato/issues/261.

Reply:
Makes sense.